toolkit/tools/internal/userutils/userutils.go

```diff
@@ -70,14 +70,21 @@ func UserExists(username string, installChroot *safechroot.Chroot) (bool, error)
 	return userExists, nil
 }
 
-func AddUser(username string, hashedPassword string, uid string, installChroot *safechroot.Chroot) error {
+func AddUser(username string, homeDir string, primaryGroup string, hashedPassword string, uid string, installChroot *safechroot.Chroot) error {
 	var args = []string{username, "-m"}
 	if hashedPassword != "" {
 		args = append(args, "-p", hashedPassword)
 	}
 	if uid != "" {
 		args = append(args, "-u", uid)
 	}
+	if len(homeDir) == 0 {
+		homeDir = filepath.Join(UserHomeDirPrefix, username)
```

Comment:
Previously the code was not passing `-d` at all into `useradd`. Is there a risk that the default behavior != expliciting setting the home dir to `filepath.Join(UserHomeDirPrefix, username)`?

Reply:
_And as a minor code reading question, comparing a string against `""` seems to be idiomatic golang, or at least common to this code base; if we do end up needing this conditional, do you think it would be more consistent to change from `len(homeDir) == 0` to `homeDir == ""`?_

Reply:
right, I don't think home dir needs to be initialized with a default. I've updated this part to only set the home dir if it's set.